backend/auth_routes.py

```diff
@@ -21,10 +21,15 @@ def decorator(*args, **kwargs):
             if not auth_token:
                 return make_response(jsonify({'message': 'authorization required for this request'}, 401))
             try:
-                # try to decode the jwt token
-                jwt_data = jwt.decode(auth_token, os.environ.get(
-                    'JWT_SALT'), algorithms=["HS256"])
+                # get env variables & check if they're there
+                jwt_salt = os.getenv('JWT_SALT')
                 jwt_secret = os.getenv("JWT_SECRET")
+                if jwt_salt is None or jwt_secret is None:
+                    return make_response(jsonify({'message': 'missing env variable'}, 404))
```

Comment:
Missing environment variables are a HTTP status code of 500 instead of 404! 
For a full list of status codes, you can check out [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)!